guiEditor/src/diagram/coordinateHelper.tsx

```diff
@@ -213,4 +219,107 @@ export class CoordinateHelper {
     public static computeLocalBounds(node: Control) {
         return new Rect(-node.widthInPixels * 0.5, -node.heightInPixels * 0.5, node.widthInPixels * 0.5, node.heightInPixels * 0.5);
     }
-} 
\ No newline at end of file
+
+    /** 
+     * converts a node's dimensions to percentage, properties can be specified as a list, or can convert all
+    */
+    public static convertToPercentage(guiControl: Control, properties?: DimensionProperties[], onPropertyChangedObservable?: Observable<PropertyChangedEvent>) {
+        let ratioX = 1;
+        let ratioY = 1;
+        if (guiControl.parent) {
+            if (guiControl.parent.typeName === "Grid") {
+                const cellInfo = (guiControl.parent as Grid).getChildCellInfo(guiControl);
+                const cell = (guiControl.parent as Grid).cells[cellInfo];
+                ratioX = cell.widthInPixels;
+                ratioY = cell.heightInPixels;
+            } else if (guiControl.parent.typeName === "Rectangle" || guiControl.parent.typeName === "Button") {
+                const thickness = (guiControl.parent as Rectangle).thickness * 2;
+                ratioX = guiControl.parent._currentMeasure.width - thickness;
+                ratioY = guiControl.parent._currentMeasure.height - thickness;
+            } else {
+                ratioX = guiControl.parent._currentMeasure.width;
+                ratioY = guiControl.parent._currentMeasure.height;
+            }
+        }
+        let old = {} as any;
+        if (!properties || properties.includes("left")) {

```

Comment:
feels like this entire section can be abstracted. Don't invest too much time for that, but see if you can prevent code repetition here. If it's not simple though, there is no need :-)

Reply:
You're totally right! Went through a few different iterations so I didn't stop to think about how simple it could ultimately be :)